server/src/routes/replays.ts

```diff
@@ -153,4 +153,35 @@ router.post('/', (req: Request, res: Response, next: Function): any => {
     req.on('error', (err) => next(err));
 });
 
+/**
+ * DELETE /replays/:replayId
+ * Deletes a replay file
+ */
+router.delete('/:replayId', async (req, res) => {
+    // Fetch replay
+    const replay = await db.getReplayById(req.params.replayId);
+    const replayUser = await db.getUserById(replay.userRef);
+
+    // Check user
+    const { user } = req;
+    if (user === undefined || user.webId !== replayUser.webId) {
+        res.status(401).send('Not authorized to delete replay.');
+        return;
+    }
+
+    await db.deleteReplayById(replay._id);
+
+    try {
+        await artefacts.deleteReplay(replay);
+    } catch (err) {
+        // If deletion failed, add the replay back into the DB
+        await db.saveReplay(replay);
```

Comment:
Why can't we use `saveReplayMetadata` here? I'd prefer not to create a separate function just for this.

Also, our error handling to keep db and objects synced is pretty bad, so nice that you thought of it here!

Reply:
Oh yeah, that does the same thing, I thought that only saved the metadata, but it also inserts the replay if it does not exist. Is that intended?

Reply:
Saving replay metadata = inserting the replay.
It doesn't do anything with the actual replay datapoints (that's all managed in the artefacts/on the route level).
If the doc contains the `_id` field, it won't assign a new one, so that should be exactly what we need.